src/sgame/sg_spawn_mover.cpp

```diff
@@ -1102,6 +1102,11 @@ void BinaryMover_act( gentity_t *ent, gentity_t *other, gentity_t *activator )
 		// open areaportal
 		if ( ent->groupMaster == ent || !ent->groupMaster )
 			trap_AdjustAreaPortalState( ent, true );
+
+		//HACK: this is really ugly, should have specialise code,
```

Comment:
Is the hack that it's set here instead of when spawning the entity? Why is it hard to set it upon spawning?

Reply:
No. It's more related to the fact that I think damage-entities should probably have their very own functions, instead of reusing this one. Here, the affectation is to *reset* when the door/button goes back to it's original position.

> Why is it hard to set it upon spawning?

It's already done on spawning, this one is a reset, not a set. Or maybe I do not understand what you mean?

Reply:
Sorry the 2nd question was just an incorrect guess. Where does the idea that the health is supposed to reset under some circumstances come from though?

Reply:
If I do not reset the health, then doors will still have negative health when closing anew. This means that the 50HP (for example) value will only be valid the 1st time they're hit.